hash_tables.py

```diff
@@ -0,0 +1,64 @@
+class HashTable(object):
```

Comment:
I THINK there's something wrong here. Here's some advice about debugging this:

1) fill in the types. 

`self.array: List[List[?]]`

2) use a debugger and step through what is happening. I think you'll notice that there's a mistake somewhere. Try inserting the same key multiple times and